### ISSUE TITLE
Check HSTS header on http response

### ISSUE THREAD

**Issue by J0WI** [ssllabs/ssllabs-scan#236]
Would be great to have a check, if the HSTS header is also send over http (which is wrong).
See https://tools.ietf.org/html/rfc6797#section-7.2

**Comment by selecadm** [comment]
SSL Labs currently doesn't scan HTTP.

https://securityheaders.io/ by @ScottHelme checks it.

The following message is shown in case of HSTS header over HTTP:
> Strict-Transport-Security: HTTP Strict Transport Security is an excellent feature to support on your site and strengthens your implementation of TLS. That said, the HSTS header should not be returned over a HTTP connection, only HTTPS.

**Comment by ivanr** [comment]
I agree that SSL Labs should be checking for HSTS set on a HTTP response. However, there's little benefit of discovering this issue because browsers won't accept such policies.

**Comment by ScottHelme** [comment]
What Ivan refers to is defined here: https://tools.ietf.org/html/rfc6797#section-8.1

"If an HTTP response is received over insecure transport, the UA MUST ignore any present STS header field(s)."

**Comment by J0WI** [comment]
It's a bad practise and sadly some admins thinks, that the HSTS header replaces the 301 redirect.

**Comment by selecadm** [comment]
@J0WI indeed: https://twitter.com/selecadm/status/663373899690852352
